src/proof/theory_proof.cpp

```diff
@@ -1021,31 +1021,30 @@ void TheoryProof::printTheoryLemmaProof(std::vector<Expr>& lemma,
   th->check(theory::Theory::EFFORT_FULL);
   Debug("pf::tp") << "TheoryProof::printTheoryLemmaProof - th->check() DONE" << std::endl;
 
-  if(oc.d_conflict.isNull()) {
+  if(oc.getLastConflict().isNull()) {
     Trace("pf::tp") << "; conflict is null" << std::endl;
-    Assert(!oc.d_lemma.isNull());
-    Trace("pf::tp") << "; ++ but got lemma: " << oc.d_lemma << std::endl;
+    Node last_lemma  = oc.getLastLemma();
+    Assert(!last_lemma.isNull());
+    Trace("pf::tp") << "; ++ but got lemma: " << last_lemma << std::endl;
 
-    if (oc.d_lemma.getKind() == kind::OR) {
+    if (last_lemma.getKind() == kind::OR) {
       Debug("pf::tp") << "OR lemma. Negating each child separately" << std::endl;
-      for (unsigned i = 0; i < oc.d_lemma.getNumChildren(); ++i) {
-        if (oc.d_lemma[i].getKind() == kind::NOT) {
-          Trace("pf::tp") << ";     asserting fact: " << oc.d_lemma[i][0] << std::endl;
-          th->assertFact(oc.d_lemma[i][0], false);
+      for (unsigned i = 0; i < last_lemma.getNumChildren(); ++i) {
+        if (last_lemma[i].getKind() == kind::NOT) {
+          Trace("pf::tp") << ";     asserting fact: " << last_lemma[i][0] << std::endl;
+          th->assertFact(last_lemma[i][0], false);
         }
         else {
-          Trace("pf::tp") << ";     asserting fact: " << oc.d_lemma[i].notNode() << std::endl;
-          th->assertFact(oc.d_lemma[i].notNode(), false);
+          Trace("pf::tp") << ";     asserting fact: " << last_lemma[i].notNode() << std::endl;
+          th->assertFact(last_lemma[i].notNode(), false);
         }
       }
-    }
-    else {
+    } else {
       Unreachable();
-
-      Assert(oc.d_lemma.getKind() == kind::NOT);
-      Debug("pf::tp") << "NOT lemma" << std::endl;
-      Trace("pf::tp") << ";     asserting fact: " << oc.d_lemma[0] << std::endl;
-      th->assertFact(oc.d_lemma[0], false);
+      // Assert(oc.d_lemma.getKind() == kind::NOT);
```

Comment:
Remove completely instead of commenting it out?

Reply:
Reverted. Eliminating dead code is out of scope.